File: EngagementHub/TOS.EngagementHub.Models/Role.cs
using System;
using System.Collections.Generic;

namespace TOS.EngagementHub.Models
{
    public class Role : EngagementModel
    {
        public string Name { get; set; }
        public Guid ProjectId { get; set; }
        public string Description { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public ICollection<RoleSkill> Skills { get; set; }
    }
}

File: NiceReads/Backend/Server/tos-server/src/application/models/Role.ts
import { Uuid } from '../../common/uuid';
import { Document } from './Document';
import { RoleSkill } from './RoleSkill';

export class Role extends Document {
    constructor(
        public name: string = '',
        public projectId: Uuid = Uuid.EMPTY,
        public description: string = '',
        public startDate: Date | undefined = undefined,
        public endDate: Date | undefined = undefined,
        public skills: RoleSkill[] = [],
        id: Uuid | undefined = undefined
    ) {
        super(id);
    }
}